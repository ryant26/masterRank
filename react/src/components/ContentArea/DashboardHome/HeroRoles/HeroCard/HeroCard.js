import React, {
  Component
} from 'react';

import HeroStatsList from '../HeroStatsList/HeroStatsList';
import HeroImage from '../../../../HeroImage/HeroImage';
import PropTypes from 'prop-types';
import Modal from '../../../../Modal/Modal';
import Model from '../../../../../model/model';

const classNames = require('classnames');


export default class HeroCard extends Component {

    constructor(props) {
        super(props);
        this.state = {
            isStatsToggleOn: false,
            modalOpen: false,
            invitable: props.hero.platformDisplayName !== props.user.platformDisplayName
        };

        this.toggleStats = this.toggleStats.bind(this);
        this.openModal = this.openModal.bind(this);
        this.closeModal = this.closeModal.bind(this);
        this.invitePlayer = this.invitePlayer.bind(this);
    }

    toggleStats() {
        this.setState(prevState => ({
            isStatsToggleOn: !prevState.isStatsToggleOn
        }));
    }

    invitePlayer() {
        const userObject = {
            "platformDisplayName": this.props.hero.platformDisplayName,
            "heroName": this.props.hero.heroName
        };
        Model.inviteUserToGroup(userObject);
    }

    openModal() {
        this.setState(() => {
            return {
                modalOpen: true
            };
        });
    }

    closeModal() {
        this.setState(() => {
            return {
                modalOpen: false
            };
        });
    }

    render() {
        const classses = classNames({
            overlay: true,
            invitable: this.state.invitable,
        });

        let statLine;

        if(this.props.hero.stats) {
            let wins = this.props.hero.stats.wins;
            let losses = this.props.hero.stats.losses;
            let winPercentage = parseFloat(wins/(wins+losses) * 100.0).toFixed(1);
            statLine = (
                <div className="sub-title">
                    <span>{this.props.hero.skillRating} SR</span>
                    <span> | </span><span>{winPercentage}% WR</span>
                </div>
            );
        }


    return (
        <div className="HeroCard flex align-center">
            <div className="invitePlayerButton">
                <HeroImage heroName={this.props.hero.heroName}/>
                <div className={classses}>
                    <div className="plus-container flex align-center justify-center" onClick={this.invitePlayer}>
                        <i className="fa fa-plus"/>
                    </div>
                </div>
            </div>
            <div className="content flex flex-column">
                <div className="display-name">{this.props.hero.platformDisplayName}</div>
                {statLine}
            </div>
            <div className="button-primary" onClick={this.openModal}>
                <div className="button-content">
                    stats
                </div>
            </div>
            <Modal modalOpen={this.state.modalOpen} closeModal={this.closeModal}>
                <HeroStatsList hero={this.props.hero} invitable={this.state.invitable}/>
            </Modal>

        </div>
    );
    }
}

HeroCard.propTypes = {
    hero: PropTypes.shape({
        heroName: PropTypes.string.isRequired,
        platformDisplayName: PropTypes.string.isRequired,
        skillRating: PropTypes.number.isRequired,
        stats: PropTypes.shape({
            wins: PropTypes.number.isRequired,
            losses: PropTypes.number.isRequired,
        })
    }),
    user: PropTypes.shape({
        platformDisplayName: PropTypes.string.isRequired
    })
};