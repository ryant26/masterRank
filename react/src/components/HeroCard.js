import React, {
  Component
} from 'react';

import HeroCardStats from './HeroCardStats.js'
import HeroImage from './HeroImage.js'
import InvitePlayerButton from './InvitePlayerButton.js'

export default class HeroCard extends Component {

  constructor(props) {
    super(props);
    this.state = {
      isStatsToggleOn: false,
      isHovering: false,
    };

    this.toggleStats = this.toggleStats.bind(this);
    this.toggleInvitePlayerButton = this.toggleInvitePlayerButton.bind(this);
  }

  toggleStats() {
    this.setState(prevState => ({
      isStatsToggleOn: !prevState.isStatsToggleOn
    }))
  }

  toggleInvitePlayerButton() {
    this.setState(prevState => ({
      isHovering: !prevState.isHovering
    }))
  }

  render() {
    const containerStyle = {
      'display': 'flex',
      'backgroundColor': '#66ccff',
      'margin': '10px',
    }

    const statsStyle = {
      'display': 'flex',
    }

    const componentStyle = {
      'textAlign': 'center',
      'height': '60px',
      'width': '33%',
    }

    let winPercentage = parseFloat(this.props.hero.general_stats.win_percentage * 100.0).toFixed(1);

    return (
      <div>
        <div style={containerStyle}
          onMouseEnter={this.toggleInvitePlayerButton}
          onMouseLeave={this.toggleInvitePlayerButton}>

          {
            this.state.isHovering
            ? <InvitePlayerButton />
            : <HeroImage heroName={this.props.hero.name}/>
          }

          <div> Username </div>

          <div style={statsStyle} onClick={this.toggleStats}>
            <div style={componentStyle} >
              <div>Rank</div>
              <div>2734</div>
            </div>

            <div style={componentStyle}>
              <div>Games Played</div>
              <div>{this.props.hero.general_stats.games_played}</div>
            </div>

            <div style={componentStyle}>
              <div>WIN %</div>
              <div>{winPercentage}%</div>
            </div>
          </div>
        </div>
        
        {
          this.state.isStatsToggleOn
          ? <HeroCardStats hero={this.props.hero}/>
          : null
        }
      </div>
    );
  }
}
