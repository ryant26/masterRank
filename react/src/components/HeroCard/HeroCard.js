import React, {
  Component
} from 'react';

import HeroCardStats from '../HeroCardStats/HeroCardStats';
import HeroImage from '../HeroImage/HeroImage';
import InvitePlayerButton from '../InvitePlayerButton/InvitePlayerButton';
import PropTypes from 'prop-types';

export default class HeroCard extends Component {

  constructor(props) {
    super(props);
    this.state = {
      isStatsToggleOn: false,
      isHovering: false,
    };

    this.toggleStats = this.toggleStats.bind(this);
    this.toggleInvitePlayerButton = this.toggleInvitePlayerButton.bind(this);
  }

  toggleStats() {
    this.setState(prevState => ({
      isStatsToggleOn: !prevState.isStatsToggleOn
    }));
  }

  toggleInvitePlayerButton() {
    this.setState(prevState => ({
      isHovering: !prevState.isHovering
    }));
  }

  render() {
    const backgroundStyle = {
      'backgroundColor': '#66ccff'      
    };

    const containerStyle = {
      'display': 'flex',
    };

    const statsStyle = {
      'display': 'flex',
      'flexDirection':'column',
      'width':'240px',
      'cursor':'s-resize'      
    };

    const componentStyle = {
      'height':'20px'
    };
    
    let wins = this.props.hero.wins;
    let losses = this.props.hero.losses;
    let winPercentage = parseFloat(wins/(wins+losses) * 100.0).toFixed(1);

    return (
      <div>
        <div style={backgroundStyle}>
          <div onMouseEnter={this.toggleInvitePlayerButton}
               onMouseLeave={this.toggleInvitePlayerButton}>
                {
                    this.state.isHovering? <InvitePlayerButton /> : 
                                           <HeroImage heroName={this.props.hero.heroName}/>
                }
          </div>
          <div style={containerStyle}>
            <div style={statsStyle} onClick={this.toggleStats}>
              <div>
                <div><b>{this.props.hero.platformDisplayName}</b></div>
              </div>
              <div style={componentStyle} >
                <div><b>Rank:</b> 2733</div>
              </div>

              <div style={componentStyle}>
                <div><b>Hours Played:</b> {this.props.hero.hoursPlayed}</div>
              </div>

              <div style={componentStyle}>
                <div><b>Win Rate:</b> {winPercentage}%</div>
              </div>
            </div>
          </div>
        </div>
        {
          this.state.isStatsToggleOn
          ? <HeroCardStats hero={this.props.hero}/>
          : null
        }
      </div>
    );
  }
}

HeroCard.propTypes = {
  hero: PropTypes.shape({
    heroName: PropTypes.string.isRequired,
    wins: PropTypes.number.isRequired,
    losses: PropTypes.number.isRequired,
    platformDisplayName: PropTypes.string.isRequired,
    hoursPlayed: PropTypes.number.isRequired,
  })
};