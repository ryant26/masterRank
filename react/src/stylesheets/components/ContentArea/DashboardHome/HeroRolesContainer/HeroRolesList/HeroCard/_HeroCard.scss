@mixin square() {
  width: 50px;
  height: 50px;
}

.HeroCard {
  position: relative;
  padding: 5px 8px;
  min-height: 60px;

  .display-name {
    margin-bottom: 4px;
    font-weight: bold;
    font-size: 14px;
    color: $gray3;
  }

  .HeroImage {
    @include square;
  }

  .content {
    padding: 10px;
  }

  .invitePlayerButton {
    overflow: hidden;
    border-radius: 3px;
  }

  .invitePlayerButton {
    .overlay {
      display: none;
      @include square;
      position: relative;
      background-color: $transparentYellow1;
      cursor: pointer;

      .plus-container {
        position: absolute;
        left: 0;
        top: 0;
        width: 100%;
        height: 100%;
        color: white;
      }
    }
  }

  .button-primary {
    display: none;
    position: absolute;
    right: 0;
    height: 15px;
    margin-right: 15px;
  }

  &:hover {
    background-color: $gray9;

    .invitePlayerButton {
      .overlay.invitable {
        display: block;
      }
    }

    .button-primary {
      display: block;
    }
  }
}