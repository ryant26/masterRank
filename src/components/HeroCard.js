import React, {
  Component
} from 'react';
import FontAwesome from 'react-fontawesome'

import HeroCardStats from './HeroCardStats.js'

export default class HeroCard extends Component {

  constructor(props) {
    super(props);
    this.state = {
      isStatsToggleOn: false,
      isHovering: false,
    };

    this.toggleStats = this.toggleStats.bind(this);
    this.toggleHover = this.toggleHover.bind(this);
  }

  toggleStats() {
    this.setState(prevState => ({
      isStatsToggleOn: !prevState.isStatsToggleOn
    }))
  }

  toggleHover() {
    this.setState(prevState => ({
      isHovering: !prevState.isHovering
    }))
  }

  render() {
    const containerStyle = {
      'display': 'flex',
      'backgroundColor': '#66ccff',
      'width': '720px',
      'margin': '10px',
    }

    const heroIconStyle = {
      'backgroundColor': 'lightgrey',
      'color': 'darkgrey',
      'height': '60px',
      'width': '60px',
      'float': 'left',
      'fontSize': '48px',
    }

    const boxStyle = {
      'textAlign': 'center',
      'height': '60px',
      'width': '25%',
    }

    let winPercentage = parseFloat(this.props.hero.general_stats.win_percentage * 100.0).toFixed(1);

    return (
      <div>
        <div style={containerStyle}
          onClick={this.toggleStats}
          onMouseEnter={this.toggleHover}
          onMouseLeave={this.toggleHover}>

          <div style={boxStyle}>
            {
              this.state.isHovering
              ? <button style={heroIconStyle}>
                  <FontAwesome name='plus' />
                </button>
              : <img style={heroIconStyle}
                  src={require(`../assets/${this.props.hero.name}-icon.png`)}
                  alt = {this.props.hero.name+' icon'}
                />
            }
          </div>

          <div style={boxStyle}>
            <div>Rank</div>
            <div>2734</div>
          </div>

          <div style={boxStyle}>
            <div>Games Played</div>
            <div>{this.props.hero.general_stats.games_played}</div>
          </div>

          <div style={boxStyle}>
            <div>WIN %</div>
            <div>{winPercentage}%</div>
          </div>
        </div>
        {
          this.state.isStatsToggleOn
          ? <HeroCardStats hero={this.props.hero}/>
          : null
        }
      </div>
    );
  }
}
