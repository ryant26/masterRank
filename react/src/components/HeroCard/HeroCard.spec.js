import React from 'react';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import HEROS from '../../resources/heroes';

import HeroCard from './HeroCard';

describe('HeroCard Component',()=> {
    let getHeroConfig = function() {
        return HEROS[0];
    };

    it('should render without exploding', () => {
        const wrapper = mount(
            <HeroCard hero={getHeroConfig()}/>
        );

        const HeroCardComponent = wrapper.find(HeroCard);
        expect(HeroCardComponent).toBeTruthy();
    });

    it('should toggle the stats card on click', () => {
        const component = renderer.create(
            <HeroCard hero={getHeroConfig()}/>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        component.getInstance().toggleStats();  
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        component.getInstance().toggleStats();  
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();

    });
});