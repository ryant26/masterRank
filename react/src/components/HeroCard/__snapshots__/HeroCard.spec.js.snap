// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroCard Component should toggle the stats card on click 1`] = `
<div>
  <div
    style={
      Object {
        "backgroundColor": "#66ccff",
      }
    }
  >
    <div
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <img
        alt="soldier76 icon"
        src={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "backgroundColor": "lightgrey",
            "float": "left",
            "height": "60px",
            "width": "60px",
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <div
        onClick={[Function]}
        style={
          Object {
            "cursor": "s-resize",
            "display": "flex",
            "flexDirection": "column",
            "width": "240px",
          }
        }
      >
        <div>
          <div>
            <b>
              PwNShoPP#1662
            </b>
          </div>
        </div>
        <div
          style={
            Object {
              "height": "20px",
            }
          }
        >
          <div>
            <b>
              Rank:
            </b>
             2733
          </div>
        </div>
        <div
          style={
            Object {
              "height": "20px",
            }
          }
        >
          <div>
            <b>
              Hours Played:
            </b>
             
            4
          </div>
        </div>
        <div
          style={
            Object {
              "height": "20px",
            }
          }
        >
          <div>
            <b>
              Win Rate:
            </b>
             
            90.9
            %
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`HeroCard Component should toggle the stats card on click 2`] = `
<div>
  <div
    style={
      Object {
        "backgroundColor": "#66ccff",
      }
    }
  >
    <div
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <img
        alt="soldier76 icon"
        src={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "backgroundColor": "lightgrey",
            "float": "left",
            "height": "60px",
            "width": "60px",
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <div
        onClick={[Function]}
        style={
          Object {
            "cursor": "s-resize",
            "display": "flex",
            "flexDirection": "column",
            "width": "240px",
          }
        }
      >
        <div>
          <div>
            <b>
              PwNShoPP#1662
            </b>
          </div>
        </div>
        <div
          style={
            Object {
              "height": "20px",
            }
          }
        >
          <div>
            <b>
              Rank:
            </b>
             2733
          </div>
        </div>
        <div
          style={
            Object {
              "height": "20px",
            }
          }
        >
          <div>
            <b>
              Hours Played:
            </b>
             
            4
          </div>
        </div>
        <div
          style={
            Object {
              "height": "20px",
            }
          }
        >
          <div>
            <b>
              Win Rate:
            </b>
             
            90.9
            %
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      style={
        Object {
          "height": "56px",
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "white",
            "float": "left",
            "height": "46px",
            "textAlign": "left",
            "width": "120px",
          }
        }
      >
        <div>
          9.16
        </div>
        <div
          style={
            Object {
              "backgroundColor": "lightgrey",
              "height": "5px",
              "width": "90px",
            }
          }
        >
          <div
            style={
              Object {
                "backgroundColor": "blue",
                "height": "5px",
                "width": "50%",
              }
            }
          />
        </div>
        <div>
          Eliminations
        </div>
      </div>
      <div
        style={
          Object {
            "backgroundColor": "white",
            "float": "left",
            "height": "46px",
            "textAlign": "left",
            "width": "120px",
          }
        }
      >
        <div>
          3.26
        </div>
        <div
          style={
            Object {
              "backgroundColor": "lightgrey",
              "height": "5px",
              "width": "90px",
            }
          }
        >
          <div
            style={
              Object {
                "backgroundColor": "blue",
                "height": "5px",
                "width": "75%",
              }
            }
          />
        </div>
        <div>
          K/D ratio
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "height": "56px",
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "white",
            "float": "left",
            "height": "46px",
            "textAlign": "left",
            "width": "120px",
          }
        }
      >
        <div>
          43
        </div>
        <div
          style={
            Object {
              "backgroundColor": "lightgrey",
              "height": "5px",
              "width": "90px",
            }
          }
        >
          <div
            style={
              Object {
                "backgroundColor": "blue",
                "height": "5px",
                "width": "60%",
              }
            }
          />
        </div>
        <div>
          Accuracy
        </div>
      </div>
      <div
        style={
          Object {
            "backgroundColor": "white",
            "float": "left",
            "height": "46px",
            "textAlign": "left",
            "width": "120px",
          }
        }
      >
        <div>
          0
        </div>
        <div
          style={
            Object {
              "backgroundColor": "lightgrey",
              "height": "5px",
              "width": "90px",
            }
          }
        >
          <div
            style={
              Object {
                "backgroundColor": "blue",
                "height": "5px",
                "width": "0%",
              }
            }
          />
        </div>
        <div>
          Blocked / Min
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "height": "56px",
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "white",
            "float": "left",
            "height": "46px",
            "textAlign": "left",
            "width": "120px",
          }
        }
      >
        <div>
          0
        </div>
        <div
          style={
            Object {
              "backgroundColor": "lightgrey",
              "height": "5px",
              "width": "90px",
            }
          }
        >
          <div
            style={
              Object {
                "backgroundColor": "blue",
                "height": "5px",
                "width": "25%",
              }
            }
          />
        </div>
        <div>
          Healing / Min
        </div>
      </div>
      <div
        style={
          Object {
            "backgroundColor": "white",
            "float": "left",
            "height": "46px",
            "textAlign": "left",
            "width": "120px",
          }
        }
      >
        <div>
          1305.59
        </div>
        <div
          style={
            Object {
              "backgroundColor": "lightgrey",
              "height": "5px",
              "width": "90px",
            }
          }
        >
          <div
            style={
              Object {
                "backgroundColor": "blue",
                "height": "5px",
                "width": "90%",
              }
            }
          />
        </div>
        <div>
          Damage / Min
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "height": "56px",
        }
      }
    >
      <div
        style={
          Object {
            "backgroundColor": "white",
            "float": "left",
            "height": "46px",
            "textAlign": "left",
            "width": "120px",
          }
        }
      >
        <div>
          9.16
        </div>
        <div
          style={
            Object {
              "backgroundColor": "lightgrey",
              "height": "5px",
              "width": "90px",
            }
          }
        >
          <div
            style={
              Object {
                "backgroundColor": "blue",
                "height": "5px",
                "width": "50%",
              }
            }
          />
        </div>
        <div>
          Obj. Kills
        </div>
      </div>
      <div
        style={
          Object {
            "backgroundColor": "white",
            "float": "left",
            "height": "46px",
            "textAlign": "left",
            "width": "120px",
          }
        }
      >
        <div>
          53.32
        </div>
        <div
          style={
            Object {
              "backgroundColor": "lightgrey",
              "height": "5px",
              "width": "90px",
            }
          }
        >
          <div
            style={
              Object {
                "backgroundColor": "blue",
                "height": "5px",
                "width": "65%",
              }
            }
          />
        </div>
        <div>
          Obj. Time
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`HeroCard Component should toggle the stats card on click 3`] = `
<div>
  <div
    style={
      Object {
        "backgroundColor": "#66ccff",
      }
    }
  >
    <div
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <img
        alt="soldier76 icon"
        src={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "backgroundColor": "lightgrey",
            "float": "left",
            "height": "60px",
            "width": "60px",
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <div
        onClick={[Function]}
        style={
          Object {
            "cursor": "s-resize",
            "display": "flex",
            "flexDirection": "column",
            "width": "240px",
          }
        }
      >
        <div>
          <div>
            <b>
              PwNShoPP#1662
            </b>
          </div>
        </div>
        <div
          style={
            Object {
              "height": "20px",
            }
          }
        >
          <div>
            <b>
              Rank:
            </b>
             2733
          </div>
        </div>
        <div
          style={
            Object {
              "height": "20px",
            }
          }
        >
          <div>
            <b>
              Hours Played:
            </b>
             
            4
          </div>
        </div>
        <div
          style={
            Object {
              "height": "20px",
            }
          }
        >
          <div>
            <b>
              Win Rate:
            </b>
             
            90.9
            %
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
